src/basic/path-lookup.h

```diff
@@ -56,9 +58,19 @@ int lookup_paths_init(LookupPaths *lp, RuntimeScope scope, LookupPathsFlags flag
 int lookup_paths_init_or_warn(LookupPaths *lp, RuntimeScope scope, LookupPathsFlags flags, const char *root_dir);
 
 int xdg_user_dirs(char ***ret_config_dirs, char ***ret_data_dirs);
-int xdg_user_runtime_dir(char **ret, const char *suffix);
-int xdg_user_config_dir(char **ret, const char *suffix);
-int xdg_user_data_dir(char **ret, const char *suffix);
+
+/* We don't treat /etc/xdg/systemd in these functions as the xdg base dir spec suggests because we assume
+ * that is a link to /etc/systemd/ anyway. */
+
+static inline int xdg_user_runtime_dir(const char *suffix, char **ret) {
+        return sd_path_lookup(SD_PATH_USER_RUNTIME, suffix, ret);
```

Comment:
libsystemd API cannot be used in basic...
I noticed that by the comment in https://github.com/systemd/systemd/pull/34434/commits/bfcf48b842644c1016583ecf0c4258cb59a102bb#diff-150e77af3e1b9e0e11a2dc72bbf346e52521f4f173da87f4c4f1379167e5e5aa

Of course, this is an inline function, hence we can place this here, but cannot be used in basic. I think it is better to move this to libsystemd/sd-path/ or so.

Reply:
Ugh, yikes, I guess that's why this was initially duplicated.

> I think it is better to move this to libsystemd/sd-path/ or so.

Yeah, moving `path-lookup.c` to sd-path is probably the way forward...